ooler/ooler.py

```diff
@@ -11,71 +11,49 @@ def __init__(self, address=None, stay_connected=True, max_connection_attempts=5)
         self.address = address
         self.stay_connected = stay_connected
         self.max_connection_attempts = max_connection_attempts
-        self.requester = GATTRequester(address, False)
+        self.client = None
         self.logger = logging.getLogger(__name__)
         self.logger.setLevel(logging.DEBUG)
         self.device_temperature_unit = None
-        if self.stay_connected:
-            self.connect()
 
-    def connect(self) -> None:
+    async def connect(self) -> None:
         """Attempt to connect to the Ooler"""
-        if self.requester.is_connected():
+        if self.client and self.client.is_connected:
             return
 
-        for attempt in range(self.max_connection_attempts):
-            try:
-                self.logger.debug(f"Attempting to connect, attempt {attempt}")
-                self.requester.connect(True)
-                self.logger.info(f"Connected to {self.address}")
-                break
-            except BTIOException as exc:
-                self.logger.warning(
-                    f"Failed to connect on attempt {attempt}, got {exc}"
-                )
-                self.requester.disconnect()
-
-        if not self.requester.is_connected():
-            raise ConnectionError(
-                f"Failed to connect after {self.max_connection_attempts} goes"
-            )
-
-        self.requester.exchange_mtu(512)
-        self.uuid_map = self._get_uuid_map()
+        self.client = BleakClient(self.address)
+        self.logger.info("Attempting to connect")
+        await self.client.connect()
```

Comment:
I assume you were hitting the error described here?
https://github.com/hbldh/bleak/issues/1176

I updated my Pi to Bookworm which has Bluez 5.66 and I am able to run with this patch now!